### PR TITLE
Fix DCIM folder not showing up

### LINKED ISSUE #400: DCIM folder with photos don't appear in Qfield Project
The DCIM folder don't appear in the android Gallery or Windows Explor after fill the form with a photo.

### Reproduction Steps:

1. Copy project to Android phone in the folder Documents.
2. Open Qfield and Open Project.
3. Open Qfield try to digitize some polygon.
4. Take a Photo.
5. Open the Gallery and try to find the DCIM folder.
6. Connect the cable to computer and try to open in Windows Explorer the project folder and don't appear the DCIM folder. 

### **OS QGIS**

Windows 10 64-bits
Windows 7 64-bits

### **Equipment's and Android SO**

- Caterpillar S41 Android (7.0.0)
- Samsung S7 Android (8.0.0) Note: Only that have the correct behavior but only when check the checkbox in the right of the photo and disconnect the cable of the computer.

- Wiko Pulp Android (5.1)
- Huawei P8 Lite Android (8.0.0)

### Expected behavior:
Open the Windows Explorer and go to the Android Phone and see the folder of DCIM inside of the Qfield project folder.

### Observed behavior:

Only visible on the Android Folder Manager. Don't appear in the Android Gallery and Windows Explorer.

### Screenshots and GIFs

Screenshots: 
Don't have for now.

Project see attach in #398 . 

**QField version:** 0.11.11 Lucendro

## Additional information:

* Problem started happening recently, didn't happen in an older version of QField: Not Sure
* Problem can be reliably reproduced, doesn't happen randomly: Yes
* Problem happens with all files and projects, not only some files or projects: No

### PR DESCRIPTION
**Pull request by m-kuhn**
Fixes #400

### CODE REVIEW

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1174-cbeb97ebc486212a19c88ee3e55a4e314b3ef014-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1174-cbeb97ebc486212a19c88ee3e55a4e314b3ef014-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1174-cbeb97ebc486212a19c88ee3e55a4e314b3ef014-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1174-cbeb97ebc486212a19c88ee3e55a4e314b3ef014-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1174-bc1b8182509fd04b09cfedb70b1359d8eb7c36e7-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1174-bc1b8182509fd04b09cfedb70b1359d8eb7c36e7-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1174-bc1b8182509fd04b09cfedb70b1359d8eb7c36e7-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1174-bc1b8182509fd04b09cfedb70b1359d8eb7c36e7-x86_64.apk)

**Comment by m-kuhn** [comment]
@RealSniperKing can you test the latest build as well? It takes into account another comment from the link on stackoverflow. :crossed_fingers:

**Comment by m-kuhn** [comment]
@RealSniperKing thanks for the latest code sample. The difference I can see is that you are also calling it on individual files (here it was just called on the folder itself). Let's see if the current build does better, it should be ready in 15 minutes.

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1174-f8a107721c1ea28eca619f395d55c0867cf23610-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1174-f8a107721c1ea28eca619f395d55c0867cf23610-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1174-f8a107721c1ea28eca619f395d55c0867cf23610-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1174-f8a107721c1ea28eca619f395d55c0867cf23610-x86_64.apk)

File: android/src/ch/opengis/qfield/QFieldCameraPictureActivity.java [hunk 1]
```diff
@@ -68,7 +69,14 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         Log.d(TAG, "resultCode: "+resultCode);
 
         File result = new File(prefix+pictureFilePath);
-        result.getParentFile().mkdirs();
+        File path = result.getParentFile();
+        path.mkdirs();
+
+        // Let the android scan new media folers/files to make them visible through MTP
```
Comment (marioba) [inline android/src/ch/opengis/qfield/QFieldCameraPictureActivity.java:75]: folers -> folders

File: android/src/ch/opengis/qfield/QFieldCameraPictureActivity.java [hunk 2]
```diff
@@ -89,6 +97,11 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
             setResult(RESULT_CANCELED, intent);
         }
 
+        // Let the android scan new media folers/files to make them visible through MTP
```
Comment (marioba) [inline android/src/ch/opengis/qfield/QFieldCameraPictureActivity.java:100]: folers -> folders

File: android/src/ch/opengis/qfield/QFieldGalleryPictureActivity.java [hunk 3]
```diff
@@ -52,7 +53,14 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         Log.d(TAG, "resultCode: "+resultCode);
 
         File result = new File(prefix+pictureFilePath);
-        result.getParentFile().mkdirs();
+        File path = result.getParentFile();
+        path.mkdirs();
+
+        // Let the android scan new media folers/files to make them visible through MTP
```
Comment (marioba) [inline android/src/ch/opengis/qfield/QFieldGalleryPictureActivity.java:59]: folers -> folders

File: android/src/ch/opengis/qfield/QFieldGalleryPictureActivity.java [hunk 4]
```diff
@@ -72,6 +80,11 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
             setResult(RESULT_CANCELED, intent);
         }
 
+        // Let the android scan new media folers/files to make them visible through MTP
```
Comment (marioba) [inline android/src/ch/opengis/qfield/QFieldGalleryPictureActivity.java:83]: folers -> folders

File: android/src/ch/opengis/qfield/QFieldCameraPictureActivity.java [hunk 5]
```diff
@@ -68,7 +69,14 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         Log.d(TAG, "resultCode: "+resultCode);
 
         File result = new File(prefix+pictureFilePath);
-        result.getParentFile().mkdirs();
+        File path = result.getParentFile();
+        path.mkdirs();
+
+        // Let the android scan new media folers/files to make them visible through MTP
+        path.setExecutable(true);
```
Comment (marioba) [inline android/src/ch/opengis/qfield/QFieldCameraPictureActivity.java:76]: Why the directory should be executable?

File: android/src/ch/opengis/qfield/QFieldGalleryPictureActivity.java [hunk 6]
```diff
@@ -52,7 +53,14 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         Log.d(TAG, "resultCode: "+resultCode);
 
         File result = new File(prefix+pictureFilePath);
-        result.getParentFile().mkdirs();
+        File path = result.getParentFile();
+        path.mkdirs();
+
+        // Let the android scan new media folers/files to make them visible through MTP
+        path.setExecutable(true);
```
Comment (marioba) [inline android/src/ch/opengis/qfield/QFieldGalleryPictureActivity.java:60]: see above

File: android/src/ch/opengis/qfield/QFieldCameraPictureActivity.java (hunk 5 above)
Comment (m-kuhn) [reply android/src/ch/opengis/qfield/QFieldCameraPictureActivity.java:76]: The code is mostly taken from https://stackoverflow.com/a/30697198/2319028
I expanded from linux file permissions (e.g. https://askubuntu.com/a/862293/452931 "Directory execute bit - You need this permission if you want to: Have any access to files within the directory.")

I'm not sure we need to set permissions at all, this is just based on the above stackoverflow link.

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1174-b4fdc1be05f2d3df74d6a9126541edeb9224214c-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1174-b4fdc1be05f2d3df74d6a9126541edeb9224214c-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1174-b4fdc1be05f2d3df74d6a9126541edeb9224214c-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1174-b4fdc1be05f2d3df74d6a9126541edeb9224214c-x86_64.apk)

**Pull request merged**